added ipv4 and ipv6 validation

msivasubramaniaan opened the pull request:
Signed-off-by: msivasubramaniaan <msivasub@redhat.com>

The existing [PR](https://github.com/redhat-developer/yaml-language-server/pull/802) has some issues due to my local reabase, Hence created a new one

### What does this PR do?
Added IPv4 and IPv6 format validation on parser

### What issues does this PR fix or reference?
https://github.com/redhat-developer/vscode-yaml/issues/832

### Is it tested? How?
Yes added UT

rgrunber commented:
When I have something like :

```json
{
    "type": "array",
    "items": {
      "type": "string",
      "format": "ipv6"
    }
}
```

```yaml
[
"2001:0db8:85a3:0000:0000:8a2e:0370:7334",
"2001:0db8:85a3:0000:0000:8a2e:0370:7334"
]
```

and associate the schema with the file, one of the entires doesn't validate.

msivasubramaniaan commented:
> ```yaml
> 2001:0db8:85a3:0000:0000:8a2e:0370:7334
> ```

@rgrunber 

YAML don't show any error on the mentioned string and it validates as expected. Please refer the screenshot
![image](https://user-images.githubusercontent.com/93245779/220823383-6db3eebf-d92d-4220-9338-a6376c8e1ae4.png)

msivasubramaniaan commented:
> When I have something like :
> 
> ```json
> {
>     "type": "array",
>     "items": {
>       "type": "string",
>       "format": "ipv6"
>     }
> }
> ```
> 
> ```yaml
> [
> "2001:0db8:85a3:0000:0000:8a2e:0370:7334",
> "2001:0db8:85a3:0000:0000:8a2e:0370:7334"
> ]
> ```
> 
> and associate the schema with the file, one of the entires doesn't validate.
> 
> **Update:** This is definitely a bug. In fact the root cause is the regular expression used for the ipv4/ipv6 patterns. You've included a flag for your particular pattern that will results in some unpredictable behaviour.

@rgrunber 
Agree. That happened due to multi-line flag. Updated the regex pattern and added all the scenarios on the test case for better coverage.

rgrunber commented on src/languageservice/parser/jsonParser07.ts:
```diff
@@ -1054,6 +1062,8 @@ function validate(
         case 'time':
         case 'email':
+        case 'ipv4':
+        case 'ipv6':
```
A few lines below here where the `exec` call is made, would it be better to change to `test` ? We don't really use any of the objects from `exec`.

rgrunber commented on src/languageservice/parser/jsonParser07.ts:
```diff
@@ -57,6 +57,14 @@ export const formats = {
+    pattern: /^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}$/,
+  },
+  ipv6: {
+    errorMessage: localize('ipv4FormatWarning', 'String does not match IPv6 format.'),
```
Should this be "ipv6FormatWarning" ?

rgrunber reviewed (approved):
Overall looks good, so once the last comments are addressed, I thin this is good to squash + merge.
